base/binaryplatforms.jl

```diff
@@ -0,0 +1,999 @@
+module BinaryPlatforms
+
+export AbstractPlatform, Platform, HostPlatform, platform_dlext, tags, arch, os,
```

Comment:
```suggestion
export Platform, HostPlatform, platform_dlext, tags, arch, os,
```

Doesn't make sense to export a deprecated type. I'm also not sure other small names like `os` make sense to be exported either, or helper functions with internal-sounding names like `parse_dl_name_version`.

Reply:
While I agree with you, I think we need to keep backward-compatibility.  Should I instead only export these from `Pkg.BinaryPlatforms`?

Reply:
`BinaryPlatforms` was never an exposed user interface was it?

Reply:
It is as exposed now as it was in `Pkg`.  Packages can `using Pkg.BinaryPlatforms` and they get a bunch of stuff, just like they can `using Base.BinaryPlatforms`.  But neither are re-exported from `Base` or from `Pkg`.

Reply:
Ok, but it wasn't documented / exposed from Pkg either, right?

Reply:
I guess the jll files gotta use it though? Is that what we want to be backwards compatible with?

Reply:
Right; older JLLs, BB and BP are the main users of it I think.

Reply:
In my opinion, it would be good to know, what are the exact things that need to be exported to make old JLLs work and what are the exact features that the new JLLs need? These are the things we should not break in the future Julias. With the exception of that, I think almost everything else should be considered internal.

Reply:
> so only users that run using Base.BinaryPlatforms are going to see any of these names.

Ok, and `Base` doesn't export `BinaryPlatforms` so everything is there is subject to change anyway. Then you can indeed export whatever.